cfgov/ask_cfpb/tests/test_search.py

```diff
@@ -1,445 +1,121 @@
-import json
+from io import StringIO
 import unittest
 from unittest import mock
 
-from django.apps import apps
-from django.http import Http404, HttpRequest, QueryDict
-from django.test import TestCase, override_settings
-from django.urls import reverse
+from django.test import TestCase
+from ask_cfpb.models.search import UNSAFE_CHARACTERS
 
 from elasticsearch.exceptions import RequestError
+from search.elasticsearch_helpers import ElasticsearchTestsMixin
 
-from ask_cfpb.documents import AnswerPageDocument
-from ask_cfpb.models import ENGLISH_PARENT_SLUG, SPANISH_PARENT_SLUG
+from ask_cfpb.forms import AutocompleteForm
+from ask_cfpb.models.answer_page import AnswerPage
 from ask_cfpb.models.search import AnswerPageSearch, make_safe
-from ask_cfpb.views import ask_search, redirect_ask_search
-from v1.util.migrations import get_or_create_page
 
 
-class AskSearchSafetyTestCase(unittest.TestCase):
+class TestSearchMakeSafe(unittest.TestCase):
     def test_make_safe(self):
-        test_phrase = "Would you like green eggs and ^~`[]#<>;|%\\{\\}\\?"
-        self.assertEqual(
-            make_safe(test_phrase), "Would you like green eggs and ?"
-        )
+        term = "What is a mortgage"
+        unsafe_term = term + "".join(UNSAFE_CHARACTERS)
+        self.assertEqual(term, make_safe(unsafe_term))
 
 
-class AnswerPageSearchTest(TestCase):
+class TestAnswerPageSearch(ElasticsearchTestsMixin, TestCase):
     def setUp(self):
         from v1.models import HomePage
 
         self.ROOT_PAGE = HomePage.objects.get(slug="cfgov")
-        self.english_parent_page = get_or_create_page(
-            apps,
-            "ask_cfpb",
-            "AnswerLandingPage",
-            "Ask CFPB",
-            ENGLISH_PARENT_SLUG,
-            self.ROOT_PAGE,
-            language="en",
-            live=True,
-        )
-        self.spanish_parent_page = get_or_create_page(
-            apps,
-            "ask_cfpb",
-            "AnswerLandingPage",
-            "Obtener respuestas",
-            SPANISH_PARENT_SLUG,
-            self.ROOT_PAGE,
-            language="es",
+
+    def test_AnswerPageSearch_init_en(self):
+        search_term = "tax"
+        english_answerPage = AnswerPageSearch(search_term=search_term)
+        assert english_answerPage.language == "en"
+        assert english_answerPage.search_term == search_term
+        assert english_answerPage.base_query == None
+        assert english_answerPage.suggestion == None
+        assert english_answerPage.results == []
```

Comment:
The Python unittest [`TestCase` class](https://docs.python.org/3/library/unittest.html#test-cases) provides a bunch of methods that capture more information around failures than a bare `assert`. These can be rewritten with those methods as:

```suggestion
        self.assertEqual(english_answerPage.language, "en")
        self.assertEqual(english_answerPage.search_term, search_term)
        self.assertIsNone(english_answerPage.base_query)
        self.assertIsNone(english_answerPage.suggestion)
        self.assertEqual(english_answerPage.results, [])
```